src/Popover/Arrow.js

```diff
@@ -1,79 +1,91 @@
-                    ${colors.white};
-                box-shadow: -1px 1px 1px 0 rgba(64, 75, 90, 0.08),
-                    -3px 3px 8px -6px rgba(64, 75, 90, 0.15);
-            }
-
-            :global([data-popper-placement^='top']) div::after {
-                transform: rotate(-45deg);
-            }
-
-            :global([data-popper-placement^='bottom']) div::after {
-                transform: rotate(135deg);
-            }
-
-            :global([data-popper-placement^='left']) div::after {
-                transform: rotate(-135deg);
-            }
-
-            :global([data-popper-placement^='right']) div::after {
-                transform: rotate(45deg);
-            }
-        `}</style>
-    </div>
-)
+const Arrow = forwardRef(({ attributes, styles }, ref) => {
+    const hidden = attributes.arrow && attributes.arrow['data-arrow-hidden']
+    const arrowPosition = getArrowPosition(attributes.popper)
+
+    return (
+        <div
+            data-test="dhis2-uicore-popoverarrow"
+            className={cx(arrowPosition, { hidden })}
+            style={styles}
+            ref={ref}
+        >
+            <style jsx>{`
+                div {
+                    width: ${ARROW_SIZE}px;
+                    height: ${ARROW_SIZE}px;
+                }
+
+                div.top {
+                    top: -${ARROW_SIZE / 2}px;
+                }
+
+                div.right {
+                    right: -${ARROW_SIZE / 2}px;
+                }
+
+                div.bottom {
+                    bottom: -${ARROW_SIZE / 2}px;
+                }
+
+                div.left {
+                    left: -${ARROW_SIZE / 2}px;
+                }
+
+                div.hidden {
+                    visibility: hidden;
+                }
+
+                div::after {
+                    content: '';
+                    position: absolute;
+                    pointer-events: none;
+                    box-sizing: border-box;
+                    border-style: solid;
+                    border-width: ${ARROW_SIZE / 2}px;
+                    border-color: transparent transparent ${colors.white}
+                        ${colors.white};
+                    box-shadow: -1px 1px 1px 0 rgba(64, 75, 90, 0.08),
+                        -3px 3px 8px -6px rgba(64, 75, 90, 0.15);
+                }
+
+                div.bottom::after {
+                    transform: rotate(-45deg);
+                }
+
+                div.top::after {
+                    transform: rotate(135deg);
+                }
+
+                div.right::after {
+                    transform: rotate(-135deg);
+                }
+
+                div.left::after {
+                    transform: rotate(45deg);
+                }
+            `}</style>
+        </div>
+    )
+})
+Arrow.displayName = 'Arrow'
 
 Arrow.propTypes = {
-    className: propTypes.string,
+    attributes: propTypes.object,
```

Comment:
should we give this a `shape` with `arrow` and `popper` keys and an optional `arrow.data-arrow-hidden`?

Reply:
(possibly without `arrow`, see below)

Reply:
Ultimately the arrow needs these two:
- `attributes.popper['data-popper-placement']`
- `attributes.arrow['data-arrow-hidden']`

Both `attributes.popper` and `attributes.arrow` can be undefined (during the first render cycle).

So currently I decided to just pass the `attributes` to the arrow and process them there.

Instead I could also use these two props:
- `popperPlacement`
- `hidden`

And effectively this means moving some complexity from the Arrow to the Popover.

Would you prefer this?
```jsx
<Arrow
    hidden={attributes.arrow && attributes.arrow['data-arrow-hidden']}
    popperPlacement={attributes.popper && attributes.popper['data-popper-placement']}
    ref={setArrowElement}
    styles={styles.arrow}
/>
```

Reply:
OK, but don't we control `attributes.arrow.data-arrow-hidden` [here (line 23)](https://github.com/dhis2/ui-core/blob/76ffa937f3a2211e9f4c4daa3c40311d7fa3b3a0/src/Popover/modifiers.js#L23) and only use it [here](https://github.com/dhis2/ui-core/blob/76ffa937f3a2211e9f4c4daa3c40311d7fa3b3a0/src/Popover/Arrow.js#L11) ?  Maybe it can also be set by `react-popper` but that's not really clear to me here, so I think I'd advocate for colocating the parsing of the hook response with the hook call, and a simple primitive props to our components.

Reply:
Implemented with primitives in https://github.com/dhis2/ui-core/pull/886/commits/4eaafb66220bf396f2feb1c8da06c101dad60742. Can this now be resolved?